src/providers/response-types/kaltura-flavor-asset-list-response.ts

```diff
@@ -10,29 +10,43 @@ export class KalturaFlavorAssetListResponse extends BaseServiceResult {
     if (!this.hasError) {
       this.totalCount = responseObj.totalCount;
       if (this.totalCount! > 0) {
-        this.data = this.setDataWithoutDuplicates(responseObj.objects);
+        this.data = this._setData(responseObj.objects);
       }
     }
   }
 
-  private setDataWithoutDuplicates(flavors: Array<any>): Array<KalturaFlavorAsset> {
-    const data = [];
-    // sort flavors by height and filter out flavors with height 0
+  private _sortFlavors = (flavors: Array<any>) => {
     const sortedFlavors = flavors.sort((a: {height: number}, b: {height: number}) => {
       return b.height - a.height;
     });
-    data.push(new KalturaFlavorAsset(sortedFlavors[0]));
-    let previousFlavor = data[0];
+    return sortedFlavors;
+  };
 
-    for (let i = 1; i < sortedFlavors.length; i++) {
-      const currentFlavor = sortedFlavors[i];
-      // Excluding audio file from filter (it's always 0)
-      if (previousFlavor.height !== currentFlavor.height && currentFlavor.height > 0) {
-        data.push(new KalturaFlavorAsset(currentFlavor));
+  private _filterFlavors = (flavors: Array<any>) => {
+    const flavorsMap = new Map();
+    flavors.forEach(flavor => {
+      if (flavor.status === 2) {
```

Comment:
extract the status to a const, otherwise the meaning of the number 2 is not as clear